tezos/new_context/src/lib.rs

```diff
@@ -347,3 +324,37 @@ mod tests {
         );
     }
 }
+
+#[derive(Default)]
+pub struct NoHash(u64);
+
+impl std::hash::BuildHasher for NoHash {
+    type Hasher = Self;
+    fn build_hasher(&self) -> Self::Hasher {
+        Self(0)
+    }
+}
+
+impl std::hash::Hasher for NoHash {
+    fn finish(&self) -> u64 {
+        self.0 as u64
+    }
+    fn write(&mut self, _slice: &[u8]) {
+        panic!("Wrong use of NoHash");
+    }
+    fn write_usize(&mut self, n: usize) {
+        self.0 = n as u64
+    }
+    fn write_u64(&mut self, n: u64) {
+        self.0 = n;
+    }
+    fn write_u32(&mut self, n: u32) {
+        self.0 = n as u64
+    }
+    fn write_u16(&mut self, n: u16) {
+        self.0 = n as u64
+    }
+}
+
+/// A map, without hashing
```

Comment:
Please add some comments explaining why this exists.

Reply:
Ok